### PR TITLE
Feat/profile enhancements

### PR DESCRIPTION
**Pull request by Darthmaul**
This PR addresses the following Trello card: https://trello.com/c/lc1Dh3J3/94-profile-loadout-listing-change-to-blade 

I have updated the user profile styling to reflect the new design, and i have stripped out the Vue components in favor of blade ones.

See images below: 
profile_pr_1
profile_pr_2

**Key Highlights:**

- Ensured "copy" and "edit" logic remained. 
- Ensured "delete" only shows if you own the loadouts
- Added backend functionality for delete on LoadoutsController -- let me know if there's any issues with how that's implemented
- Mobile styling in-place

This was a priority as a tremendous amount of our traffic actually hits these pages, we could definitely improve our audience's experience with this update.

### CODE REVIEW

File: app/Http/Controllers/LoadoutsController.php [hunk 1]
```diff
@@ -45,6 +46,21 @@ public function preview($id)
         return view('loadouts.preview')->withLoadout($loadout);
     }
 
+    public function delete($id)
```
Comment (chaseconey) [inline app/Http/Controllers/LoadoutsController.php:49]: We can use a native Request here to validate and ensure user has permissions.

File: resources/scss/app-v2.scss [hunk 2]
```diff
@@ -163,3 +163,31 @@
         }
     }
 }
+
+.profileButton {
```
Comment (chaseconey) [inline resources/scss/app-v2.scss:167]: I think we can get this done with native Tailwind. Can we try converting it over?

File: resources/views/components/profile-loadout.blade.php [hunk 3]
```diff
@@ -0,0 +1,67 @@
+<li class="flex flex-row flex-wrap">
```
Comment (chaseconey) [inline resources/views/components/profile-loadout.blade.php:1]: If this is the exact same as our dashboard, could we use the component over there as well?

**Review by chaseconey — CHANGES_REQUESTED** [review]
Reviewed on mobile but just a few things that I can see so far.

File: resources/views/components/profile-loadout.blade.php (hunk 3 above)
Comment (Darthmaul) [reply resources/views/components/profile-loadout.blade.php:1]: This one is decently customized and has some permissions stuff at the bottom, I don't think they can live in the same one.

File: app/Http/Controllers/LoadoutsController.php (hunk 1 above)
Comment (Darthmaul) [reply app/Http/Controllers/LoadoutsController.php:49]: Can you give an example? Not sure what you mean

File: resources/scss/app-v2.scss (hunk 2 above)
Comment (Darthmaul) [reply resources/scss/app-v2.scss:167]: If we're going to do that I would rather just make a new button style, didn't we want to do that anyways?

File: app/Http/Controllers/LoadoutsController.php (hunk 1 above)
Comment (chaseconey) [reply app/Http/Controllers/LoadoutsController.php:49]: https://laravel.com/docs/8.x/validation this should have some good examples for ya.

File: resources/scss/app-v2.scss (hunk 2 above)
Comment (chaseconey) [reply resources/scss/app-v2.scss:167]: Yeah I really want to re-style the button, this is probably a good time to do it.

File: resources/views/components/profile-loadout.blade.php (hunk 3 above)
Comment (chaseconey) [reply resources/views/components/profile-loadout.blade.php:1]: Can we pull the buttons out of the component and then use the actual detail view portion in both places? That should make it more portable I believe.

**Comment by chaseconey** [comment]
I also think in general, the centered with a set width is kind of an odd look - not sure what we can do instead, but making it full  width would probably be the first thing to try there.

File: resources/views/components/profile-loadout.blade.php (hunk 3 above)
Comment (Darthmaul) [reply resources/views/components/profile-loadout.blade.php:1]: Hmm...not sure how we would do this.
We call the listing component in the main view: `dashboard-listing`, which has this main utility: 
```php
@foreach($loadoutList as $loadout)
    <x-profile-loadout :loadout="$loadout"/>
@endforeach
```
This has some surrounding HTML which I would argue can be lifted out of this component, and we can scrap that component itself. This alleviates the need to have separate listing components, and allows me to focus on the main area that needs customization: the loadout card itself. Also, there are some stylings in `dashboard-listing` which I needed to manipulate in this specific case (the stylings in there are specific to "dashboard" page). 
Again, these can be lifted out of the component and just call the `dashboard-loadout` in a `foreach` loop at that level

Now, onto the `dashboard-loadout`.  I've styled the actual component itself slightly different to accommodate spacing to give the buttons breathing room.
Then there's the additional `flex-col` I need to have for my buttons themselves.
You could definitely say this belongs in a separate component, but it will be a one-off. Do we really want to pull all this out just to have some one-off components anyways? I would rather know where all the profile logic/styling lives so I can chase it all down at once. If this starts to become logic we need in a lot of other areas, we can work to separate it out -- I just see no value in it for this one scenario.

In summary:
- Recommend removing the `dashboard-listing` component to allow for a separate `profile-loadout` component to be loaded
- Do not recommend separating the different pieces of `profile-loadout` component to support singular use-case

**Comment by Darthmaul** [comment]
> 
> 
> I also think in general, the centered with a set width is kind of an odd look - not sure what we can do instead, but making it full width would probably be the first thing to try there.
Yeah I was just trying to match what we had currently. This is with full-width and enough room for the buttons:
image
I'm not opposed to that if you like it

File: resources/scss/app-v2.scss (hunk 2 above)
Comment (Darthmaul) [reply resources/scss/app-v2.scss:167]: An initial first pass. Do we want to go a different direction w/ our buttons or just keep the same aesthetic with similar styling?
Screen Shot 2021-05-17 at 10 33 19 AM

File: app/Http/Controllers/LoadoutsController.php (hunk 1 above)
Comment (Darthmaul) [reply app/Http/Controllers/LoadoutsController.php]: Let me know if I implemented what you were thinking of via this commit: https://github.com/drg-tools/karl.gg/pull/202/commits/6b671d944238c54129498b46b140e1a0ed4d6d5b

Thanks

File: resources/scss/app-v2.scss (hunk 2 above)
Comment (chaseconey) [reply resources/scss/app-v2.scss:167]: I really like this, I think we just need to add some rounded corners to match the rest of the design :).

Comment (Darthmaul) [reply resources/scss/app-v2.scss:167]: Bam done 
Screen Shot 2021-05-19 at 2 36 16 PM

Comment (Darthmaul) [reply resources/scss/app-v2.scss:167]: Changed to the karl-orange variable
Screen Shot 2021-05-19 at 2 41 17 PM
